rclcpp/src/rclcpp/publisher_base.cpp

```diff
@@ -98,6 +101,8 @@ PublisherBase::PublisherBase(
     rmw_reset_error();
     throw std::runtime_error(msg);
   }
+
+  bind_event_callbacks(event_callbacks_, use_default_callbacks);
```

Comment:
It's a bad practice to call virtual methods for the constructor, as it will call the method defined here (see https://en.cppreference.com/w/cpp/language/virtual#During_construction_and_destruction).
Anyway, there's no need for the method to be `virtual` really, so I would just remove the qualifier.

Reply:
👍 this will call base class method every time, same thing goes to dtor.